test/helpers/mock_server.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'webrick'
+
+module MockServer
+  extend self
+
+  def start(port)
+    pid = Thread.new do
+      TestServer.new(hostname: '0.0.0.0', port: port).start
+    end
+    sleep(1)
+    puts "Created test server #{pid}, port: #{port}"
+    pid
+  end
+
+  def cleanup(pid)
+    Thread.kill(pid)
+    puts "Killed test server #{pid}."
+  end
+end
+
+class TestServer
+  def initialize(hostname:, port:)
+    dev_null = WEBrick::Log.new("/dev/null", 7)
```

Comment:
Is there a way to put this into an artifact directory, so it's available from Travis?
https://docs.travis-ci.com/user/uploading-artifacts/

Reply:
Seems like we'll need to bring in our on S3 bucket for that

Reply:
Any reason to not have this output to stderr (i.e., use `nil` for the filename)? I'm assuming those logs show up in the Travis UI.

Reply:
Just wanted to avoid lots of logs in the CI, primarily the reason why I went with no logs.
Took the silent logging out in the recent commit

Reply:
If there's way too many logs, we could always have our own custom logger that logs only the interesting things like errors :)

Up to you. If you think there's way too many log lines, feel free to keep it `/dev/null`. Maybe just the access log is `/dev/null`? I'll leave all of that to your discretion.